libs/langchain/langchain/document_loaders/base_o365.py

```diff
@@ -0,0 +1,182 @@
+"""Base class for all loaders that uses O365 Package"""
+from __future__ import annotations
+
+import logging
+import os
+import tempfile
+from abc import abstractmethod
+from enum import Enum
+from pathlib import Path
+from typing import TYPE_CHECKING, Dict, Iterable, List, Sequence, Union
+
+from langchain.document_loaders.base import BaseLoader
+from langchain.document_loaders.blob_loaders.file_system import FileSystemBlobLoader
+from langchain.document_loaders.blob_loaders.schema import Blob
+from langchain.pydantic_v1 import BaseModel, BaseSettings, Field, FilePath, SecretStr
+
+if TYPE_CHECKING:
+    from O365 import Account
+    from O365.drive import Drive, Folder
+
+logger = logging.getLogger(__name__)
+
+CHUNK_SIZE = 1024 * 1024 * 5
+
+
+class _O365Settings(BaseSettings):
+    client_id: str = Field(..., env="O365_CLIENT_ID")
+    client_secret: SecretStr = Field(..., env="O365_CLIENT_SECRET")
+
+    class Config:
+        env_prefix = ""
+        case_sentive = False
+        env_file = ".env"
+
+
+class _O365TokenStorage(BaseSettings):
+    token_path: FilePath = Path.home() / ".credentials" / "o365_token.txt"
+
+
+class _FileType(str, Enum):
+    DOC = "doc"
+    DOCX = "docx"
+    PDF = "pdf"
+
+
+def fetch_mime_types(file_types: Sequence[_FileType]) -> Dict[str, str]:
+    mime_types_mapping = {}
+    for file_type in file_types:
+        if file_type.value == "doc":
+            mime_types_mapping[file_type.value] = "application/msword"
+        elif file_type.value == "docx":
+            mime_types_mapping[
+                file_type.value
+            ] = "application/vnd.openxmlformats-officedocument.wordprocessingml.document"  # noqa: E501
+        elif file_type.value == "pdf":
+            mime_types_mapping[file_type.value] = "application/pdf"
+    return mime_types_mapping
+
+
+class O365BaseLoader(BaseLoader, BaseModel):
```

Comment:
Haven't worked with O365 before. What type of stuff is accessible except for sharepoint? (Trying to understand why inheritance is needed)